src/Mod/Part/Gui/TaskAttacher.cpp

```diff
@@ -345,47 +350,194 @@ bool TaskAttacher::updatePreview()
 
 QLineEdit* TaskAttacher::getLine(unsigned idx)
 {
-    switch(idx) {
-        case 0: return ui->lineRef1;
-        case 1: return ui->lineRef2;
-        case 2: return ui->lineRef3;
-        case 3: return ui->lineRef4;
-        default: return nullptr;
+    switch (idx) {
+    case 0: return ui->lineRef1;
+    case 1: return ui->lineRef2;
+    case 2: return ui->lineRef3;
+    case 3: return ui->lineRef4;
+    default: return nullptr;
+    }
+}
+
+void TaskAttacher::processSelection(App::DocumentObject*& rootObj, std::string& sub)

```

Comment:
`processSelection` is a pretty generic name here: I suggest giving it a more verbose name ("process" doesn't really mean much in this context -- what does this method actually *do*?).